src/ffi/extra/json_parser.cc

```diff
@@ -630,7 +632,7 @@ class JSONParser {
       return true;
     }
     // non-empty object
-    while ((next_char = ctx_.Peek()) != -1) {
+    while (ctx_.Peek() != -1) {
```

Comment:
there is next char query that immediately follows this, remove next_char assignment may leads to an issue here